src/Kinesis.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Cascade\StatusStream;
+
+use Aws\Kinesis\KinesisClient;
+
+class Kinesis implements StatusStreamInterface
+{
+    private $component = '';
+    private $kinesis;
+    private $machineId;
+    private $streamName;
+    private $system = '';
+    private $subsystem = '';
+
+    public function __construct(KinesisClient $kinesis, $streamName)
+    {
+        $this->kinesis = $kinesis;
+        $this->streamName = $streamName;
+    }
+
+    /**
+     * @param string $system
+     * @param string $subsystem
+     * @param string $component
+     */
+    public function setSystemId($system, $subsystem = '', $component = '')
+    {
+        $this->system = $system;
+        $this->subsystem = $subsystem;
+        $this->component = $component;
+    }
+
+    /**
+     * @param array $machineId
+     */
+    public function setMachineId(array $machineId)
```

Comment:
Yeah, I think something minor is being missed partly because the name is not clear enough.

This is really a _unique process instance ID_, which is almost always a combination of at least EC2 instance ID and docker container ID (potentially with a process ID added as well). So the normal case here is that it's an ID with multiple parts.

Still very much thinking about changing this function though, so maybe just flattening it to a string makes sense at that point.